components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1763,12 +1792,12 @@ public void deleteGroup(String groupId) throws NotFoundException, CharonExceptio
             throw new NotImplementedException(error);
         }
 
-        if (SCIMCommonUtils.isFilteringEnhancementsEnabled()) {
+        if (isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled()) {
+            attributeValue = prependPrimaryDomain(attributeValue);
```

Comment:
This adds primary domain to every attribute. Is this necessary. Only specific group attributes could have domain prepended and handled right similar to username

Reply:
Followings are the supported attributes in Groups endpoint.

- displayName
- members
- members.value
- members.$ref
- members.display
- members.type

Previously we add the 'PRIMARY/' in-front of the 'displayName' attribute only. However when newly introduced property (FilterUsersAndGroupsOnlyFromPrimaryDomain) enabled, The 'PRIMARY/' will be added in-front of every attribute which are listed above.

Reply:
Does `prependPrimaryDomain` method handle the scenario where `attributeValue` already comes with a domain prepended?

Reply:
My question is why should it add.
domain separator can be a a character used in any other attribute that displayName and members.value.
How in such a case primary domain would be identified
The intention here is, if not domain specified (in an attribute where user store domain is allowed like displayName or as the 'domain' query param) to switch to filter only from primary domain.
The method used here is error prone, moreover, from each such attribute when filtering from the user store it self, is this prepended domain handled

Reply:
@tharindu-bandara yes, `prependPrimaryDomain` method handle the scenario where attributeValue already comes with a domain prepended. I'll change the method name.

@malithie, Currently domain  query param is not supported. Therefore, I'll  change the above to prepend 'PRIMARY/' only for  **displayName** and **members.value**. Once domain query param support for Groups endpoint. We can do the same thing what we did as Users endpoint.

Reply:
I still don't think this sorts out the problem. Also members.value seems to refer to the user id and not user name. Then again it should be members.display
Can you check the attributes and fix accordingly.
Anyhow irrespective of the attribute you have to make sure if it's set to list or filter from primary to filter only from primary.
Basically, if there's no domain specified in the domain allowed attributes, filtering from any attribute should happen from the primary domain. IMO, for groups may be the API does not support domain, but internally you can resolve the domain to list of filter from and you can list or filter from the respective user store domain

Reply:
Currently Groups endpoint not supported for domain query param. Once it is done by this PR[1],
It is possible to resolve the domain name and use the existing methods as we did in the Users endpoint.
[1][https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/166](https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/166)